controllers/azurecluster_controller.go

```diff
@@ -113,8 +113,8 @@ func (acr *AzureClusterReconciler) SetupWithManager(ctx context.Context, mgr ctr
 // +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;
 // +kubebuilder:rbac:groups=resources.azure.com,resources=resourcegroups,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=resources.azure.com,resources=resourcegroups/status,verbs=get;list;watch
-// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways/status,verbs=get;list;watch
+// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways;bastionhosts,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways;bastionhosts/status,verbs=get;list;watch
```

Comment:
Sorry again 😬
I think we want to have descriptive annotations rather than clubbing them together. Another discussion point at [this comment on PrivateEndpoint to ASO PR](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4108#discussion_r1370788398)

Reply:
I think the point I was trying to make there was separate. I don't think it matters whether or not we have separate lines per-resource or per-group.

Reply:
Nonetheless, I think we need to get `// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways;bastionhosts/status,verbs=get;list;watch
` updated to `// +kubebuilder:rbac:groups=network.azure.com,resources=natgateways/status;bastionhosts/status,verbs=get;list;watch
`